src/main/java/dao/H2Factory/H2OrderDao.java

```diff
@@ -3,173 +3,122 @@
+import org.springframework.jdbc.support.GeneratedKeyHolder;
+import org.springframework.jdbc.support.KeyHolder;
+import org.springframework.stereotype.Repository;
 
+@Repository("h2OrderDao")
 public class H2OrderDao implements OrderDaoInterface {
 
-    private final ConnectionProvider connectionProvider;
-    private final CustomerDaoInterface customerDao;
-    private static final Logger LOG = LoggerFactory.getLogger(H2OrderDao.class);
-    private final ProductConverter productConverter;
+    @Autowired
+    private NamedParameterJdbcTemplate jdbcTemplate;
 
-    protected H2OrderDao(ConnectionProvider connectionProvider,
-            CustomerDaoInterface customerDao) {
-        this.connectionProvider = connectionProvider;
-        this.customerDao = customerDao;
-        productConverter = ProductConverter.getInstance();
+    H2OrderDao() {
     }
 
-    public void create(Order newOrder) throws DaoException {
-        try {
-            PreparedStatement st = connectionProvider.getConnection()
-                    .prepareStatement("insert into customerapplication."
-                            + "ordertable(customerid,orderdate) values(?,?)");
-            st.setInt(1, newOrder.getCustomer().getCardNumber());
-            st.setDate(2, new java.sql.Date(new Date().getDate()));
-            st.executeUpdate();
-            ResultSet pk = st.getGeneratedKeys();
-            long currentId = 0;
-            if (pk.next()) {
-                currentId = pk.getLong(1);
-            }
-            for (Product product : newOrder.getProducts()) {
-                st = connectionProvider.getConnection().prepareStatement(
-                        "set @idvar = ?; insert into customerapplication"
-                        + ".product_order(orderid,"
-                        + "productid) values(@idvar,?)");
-                st.setLong(1, currentId);
-                st.setInt(2, product.getProductId());
-                st.executeUpdate();
-                connectionProvider.destroy();
-            }
-        } catch (SQLException ex) {
-            LOG.error(ex.getSQLState());
-            throw new DaoException("Inserting data error", ex);
+    public void create(Order newOrder) {
+        String queryOrderTable = "insert into customerapplication."
+                + "ordertable(customerid,orderdate) "
+                + "values(:customerid,:orderdate)";
+        MapSqlParameterSource parameters = new MapSqlParameterSource();
+        parameters.addValue("customerid", newOrder
+                .getCustomer().getCardNumber());
+        parameters.addValue("orderdate", newOrder.getOrderDate());
+        KeyHolder keyHolder = new GeneratedKeyHolder();
+        jdbcTemplate.update(queryOrderTable, parameters, keyHolder,
+                new String[]{"GENERATED_ID"});
+        Number generatedId = keyHolder.getKey();
+        int tempId = generatedId.intValue();
+        String queryProductOrder = "set @idvar = :idvar; insert into "
```

Comment:
Why do you need @idvar? it can be done in the way "insert into table(:id, :name)"